src/common/chunked.ts

```diff
@@ -296,28 +310,90 @@ export class ChunkedUploader {
       throw firstError;
     }
 
+    const finalizeResponse = await this.finalizeUpload(
+      uploadId,
+      dataItemByteCount,
+      dataItemOpts?.paidBy,
+      combinedSignal,
+    );
+
+    emitter.emit('upload-success');
+    return finalizeResponse;
+  }
+
+  private toGiB(bytes: ByteCount): ByteCount {
+    return bytes / 1024 ** 3;
+  }
+
+  private async finalizeUpload(
+    uploadId: string,
+    dataItemByteCount: ByteCount,
+    paidBy?: string | string[],
+    signal?: AbortSignal,
+  ): Promise<TurboUploadDataItemResponse> {
+    // Wait up to 1 minute per GiB of data for the upload to finalize
+    const fileSizeInGiB = Math.ceil(this.toGiB(dataItemByteCount));
+    const defaultMaxWaitTime = fileSizeInGiB;
```

Comment:
Nit: naming. Time has no unit here but it's intended to be minutes.

Reply:
included in this refactor: 992a566